HexDump/HexDump_Parse.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Drew Noakes. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace HexDump
+{
+    public  static partial class HexDump
+    {
+        /// <summary>
+        /// Parse HexDump into a byte array
+        /// </summary>
+        /// <param name="dump"></param>
+        /// <param name="columnWidth"></param>
+        /// <param name="columnCount"></param>
+        /// <param name="includeOffset"></param>
+        /// <param name="includeAscii"></param>
+        /// <returns></returns>
+        public static byte[] Parse(string dump, int columnWidth = 8, int columnCount = 2, bool includeOffset = true, bool includeAscii = true)
```

Comment:
Do you think it's possible to make parsing work without so many options?

```
00000000   01 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00   ................
```

What if the parser just looked for valid pairs of hex characters separated by whitespace?

- The offset would have to have more than two characters to be automatically excluded
- The ASCII section could contain eg `00 01 02`

The second point is the hardest one to deal with.

Reply:
basic state machine will do the trick

Reply:
Consider the case:

```
00000000   30 31 20 30 32 30 30 33 20 30 34 20 30 35 20 30   01 02 03 04 05 0
```

Reply:
ParseStateMachine takes into account special cases like this one.